src/renderer/fonts/caching_shaper.rs

```diff
@@ -79,29 +83,57 @@ impl CachingShaper {
     }
 
     fn reset_font_loader(&mut self) {
-        let font_size = self.options.size * self.scale_factor;
+        // Calculate the new fudge factor required to scale the font width to the nearest exact pixel
+        // NOTE: This temporarily loads the font without any fudge factor, since the interface
+        // nees a size and we don't know the exact one until it's calculated.
+        self.fudge_factor = 1.0;
+        let mut font_size = self.current_size();
         trace!("Using font_size: {:.2}px", font_size);
+        self.font_loader = FontLoader::new(font_size);
+        let (metrics, font_width) = self.info();
+        trace!(
+            "Font width: {:.2}px {:.2}px",
+            font_width,
+            metrics.average_width
+        );
+        self.fudge_factor = font_width.round() / font_width;
```

Comment:
@fredizzimo this `fudge_factor` is actually has broken something. Now we can't use non-exact font sizes, and for some reason, my font (JetBrains Mono, looks good only with some non-integer sizes).

And difference is pretty big, compare this two screens:

With fudging set to 1:

![image](https://user-images.githubusercontent.com/301015/151059152-2510f77b-68f9-4b78-82d2-952bb39d96b0.png)

```
TRACE [neovide::renderer::fonts::caching_shaper] Fudge factor: 1.00
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font size: 23.53px
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font width: 14.12px
```

With actual fudging:
![image](https://user-images.githubusercontent.com/301015/151059052-13d5b116-755b-4f18-baf1-49406f414acb.png)

```
TRACE [neovide::renderer::fonts::caching_shaper] Fudge factor: 0.99
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font size: 23.33px
TRACE [neovide::renderer::fonts::caching_shaper] Fudged font width: 14.00px
```

That little difference between 14.00 and 14.12 changes something in font rendering, and at 14.12 it renders properly, with correct height. (JetBrainsMono actually uses that `Increased height for better reading expirience` at ther website, and i can clearly notice when something is wrong).

I think it would be good to add option to disable this fudging. Or maybe i'll look for options to resolve issue with squashed fonts without scaling fonts to nearest integer.

Where is that rounding you talked about in pr description?

> Previously the font width was always rounded down to the nearest integer